src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -726,4 +750,93 @@ public static void constructIonStruct(IonStruct fields, IonList annotations, Ion
+                    for (int i = 0; i < listSize; i++) {
+                        occurTime = 1;
+                        WriteRandomIonValues.constructScalarTypeData(type, writer, occurTime);
+                    }
+                } else if (constraints.get(IonSchemaUtilities.KEYWORD_ORDERED_ELEMENTS) != null) {
+                    IonList orderedElement = (IonList) constraints.get(IonSchemaUtilities.KEYWORD_ORDERED_ELEMENTS);
+                    for (int index = 0; index < orderedElement.size(); index++) {
+                        IonType elementType = orderedElement.get(index).getType();
+                        IonType valueType;
+                        switch (elementType) {
+                            case SYMBOL:
+                                occurTime = 1;
+                                valueType = IonType.valueOf(orderedElement.get(index).toString().toUpperCase());
+                                WriteRandomIonValues.constructScalarTypeData(valueType, writer, occurTime);
+                                break;
+                            case STRUCT:
+                                IonStruct structConstraints = (IonStruct) orderedElement.get(index);
+                                occurTime = IonSchemaUtilities.parseConstraints(structConstraints, IonSchemaUtilities.KEYWORD_OCCURS);
+                                if(occurTime == 0) {
+                                    break;
+                                }
+                                valueType = IonType.valueOf(structConstraints.get(IonSchemaUtilities.KEYWORD_TYPE).toString().toUpperCase());
+                                WriteRandomIonValues.constructScalarTypeData(valueType, writer, occurTime);
+                                break;
+                        }
+                    }
+                }
+                writer.stepOut();
+            }
+        }
+    }
+
+    /**
+     * Construct scalar type Ion data based on the occurrence time. This method is mainly reused during the process of generating Ion List which will specify the occurrence time.
+     * @param valueType is IonType of the data needed to be written in Ion List.
+     * @param writer is IonWriter.
+     * @param occurTime is the occurrence time of the element in Ion List.
+     * @throws IOException if errors occur when writing data.
+     */
+    public static void constructScalarTypeData(IonType valueType, IonWriter writer, int occurTime) throws IOException {
+        for (int i = 0; i < occurTime; i++) {
+            switch (valueType) {
+                // If more scalar types of Ion data are supported, this is the point to add more cases.
+                case STRING:
+                    writer.writeString(WriteRandomIonValues.constructString(defaultRange, 20));
+                    break;
+                case INT:
+                    writer.writeInt(WriteRandomIonValues.constructInt());
```

Comment:
When you have a method that is known to be incomplete for certain enum values, I always recommend throwing some sort of exception if the method is provided with one of the unsupported enum values. That will make it much easier to figure out what's going on if you try to add support for other scalar types elsewhere but forget to update this method.

Reply:
> When you have a method that is known to be incomplete for certain enum values, I always recommend throwing some sort of exception if the method is provided with one of the unsupported enum values. That will make it much easier to figure out what's going on if you try to add support for other scalar types elsewhere but forget to update this method.

Added the 'default' case which can throw exceptions when the current enum type is not supported.